cpp/CMakeLists.txt

```
@@ -334,17 +334,57 @@ target_link_libraries(cugraph PRIVATE
     ${OpenMP_CXX_LIB_NAMES})
 endif()
 
+################################################################################
+# - C-API library --------------------------------------------------------------
+
+add_library(cugraph_c SHARED
+        src/c_api/cugraph_api.cpp
+        )
+add_library(cugraph::cugraph_c ALIAS cugraph_c)
 
-###################################################################################################
-# - generate tests --------------------------------------------------------------------------------
+# Currently presuming we aren't calling any CUDA kernels in cugraph_c
+
+set_target_properties(cugraph_c
+    PROPERTIES BUILD_RPATH                         "\$ORIGIN"
+               INSTALL_RPATH                       "\$ORIGIN"
+               # set target compile options
+               CXX_STANDARD                        17
+               CXX_STANDARD_REQUIRED               ON
+               INTERFACE_POSITION_INDEPENDENT_CODE ON
+)
+
+target_compile_options(cugraph_c
+            PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${CUGRAPH_CXX_FLAGS}>"
+)
+
+################################################################################
+# - C-API include paths --------------------------------------------------------
+target_include_directories(cugraph_c
+    PRIVATE
+        "${CMAKE_CURRENT_SOURCE_DIR}/src"
+    PUBLIC
+        "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+        "$<INSTALL_INTERFACE:include>"
+)
+
+################################################################################
+# - C-API link libraries -------------------------------------------------------
+target_link_libraries(cugraph_c
+        PUBLIC
+                cugraph::cugraph
+                FAISS::FAISS
```

```suggestion
                FAISS::FAISS
    PRIVATE
        CUDA::cublas
        CUDA::curand
        CUDA::cusolver
        CUDA::cusparse             
```

To resolve issues not find cublas / cusolve / etc issues

---

Lets mark FAISS private as well

---

Thanks @robertmaynard . We discussed offline and decided to also try making FAISS private. I'll push that change and make FAISS public again if it results in problems.